packages/core/src/auth/tokens/token-scope.ts

```diff
@@ -0,0 +1,35 @@
+export enum CacheRole {
+  None = 'none',
+  ReadWrite = 'readwrite',
+}
+
+export interface CacheRestriction {
+  cacheRole: CacheRole;
+}
+
+export enum TopicRole {
+  None = 'none',
+  ReadWrite = 'readwrite',
+}
+
+export interface TopicRestriction {
+  topicRole: TopicRole;
+}
+
+export type Restriction = CacheRestriction | TopicRestriction;
+
+export interface Restrictions {
```

Comment:
This feels backward to me, at least with the naming. I'd hope that we are explicitly allowing access, instead of explicitly denying access to actions. 

The way this reads is that I pass in the things that are not allowed. If I'm misreading, I think renaming to something like `Permissions` would help with intuitiveness.

Reply:
lol.  This same convo/debate was happening while they were ironing out what to name these fields on the server-side, and in this PR I just copied what they called them on the server side for now.  But we could change it back, either here or in both places.

One other thing to keep in mind Allen is that for the most part I wouldn't expect an end user to ever mess with these directly.  We would want to give them pre-built things like `AllDataReadWrite`, and composable things (in the future) like `SingleCacheReadOnly`, that they can chain together to build up a Scope w/o ever having to see the word Restriction.  Only advanced use cases should ever need to go that deep.

Also of note: the `SuperUser` role is achieved by providing an empty list of Restrictions.  So I do think that makes some sense and is why they chose the current terminology.  It would seem a little weird to create a SuperUser role by providing an empty list of permissions.  But this is another thing that we don't actually intend to surface to users because they won't have permission to create such a token for now anyway.

All that said, I have no major skin in the game in the naming choice between `Permission` and `Restriction`.

@kvcache @rlinehan do you all want to weigh in on this at all?

Reply:
Also @allenheltondev since you can't directly see it in the PR, this is what I think it will look like in your web app, to create your tokens for browsers:

```
authClient.generateAuthToken(new AllDataReadWrite(), ExpiresIn.hours(2));
```

Reply:
renamed from Restrictions to Permissions in https://github.com/momentohq/client-sdk-nodejs/pull/515/commits/477a5d4f15b06648bd5b7d226e055de793c87364